src/Microsoft.VisualStudio.Editors/PropPages/TargetFrameworkMoniker.vb

```diff
@@ -50,105 +50,57 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
             Return _displayName
         End Function
 
-        'TODO: Remove this hardcoded list (Refer Bug: #795)
-        Private Shared Function AddDotNetCoreFramework(prgSupportedFrameworks As Array, supportedTargetFrameworksDescriptor As PropertyDescriptor) As Array
-            Dim _TypeConverter As TypeConverter = supportedTargetFrameworksDescriptor.Converter
-            If _TypeConverter IsNot Nothing Then
-                Dim supportedFrameworksList As List(Of String) = New List(Of String)
-                For Each moniker As String In prgSupportedFrameworks
-                    supportedFrameworksList.Add(moniker)
-                Next
-
-                For Each frameworkValue In _TypeConverter.GetStandardValues()
-                    Dim framework = CStr(frameworkValue)
-                    If framework IsNot Nothing Then
-                        supportedFrameworksList.Add(framework)
-                    End If
-                Next
-
-                Return supportedFrameworksList.ToArray
-            End If
-
-            Return prgSupportedFrameworks
-        End Function
-
         ''' <summary>
         ''' Gets the supported target framework monikers from DTAR
         ''' </summary>
-        ''' <param name="vsFrameworkMultiTargeting"></param>
         Public Shared Function GetSupportedTargetFrameworkMonikers(
-            vsFrameworkMultiTargeting As IVsFrameworkMultiTargeting,
-            currentProject As Project,
-            supportedTargetFrameworksDescriptor As PropertyDescriptor) As IEnumerable(Of TargetFrameworkMoniker)
-
-            Dim supportedFrameworksArray As Array = Nothing
-            VSErrorHandler.ThrowOnFailure(vsFrameworkMultiTargeting.GetSupportedFrameworks(supportedFrameworksArray))
-            If supportedTargetFrameworksDescriptor IsNot Nothing Then
-                supportedFrameworksArray = AddDotNetCoreFramework(supportedFrameworksArray, supportedTargetFrameworksDescriptor)
+                frameworkMultiTargeting As IVsFrameworkMultiTargeting,
+                project As Project,
+                converter As TypeConverter) As IEnumerable(Of TargetFrameworkMoniker)
+
+            Dim provider = GetSupportedTargetFrameworksProvider(frameworkMultiTargeting, project, converter)
+
+            Dim moniker As [Property] = project.Properties.Item(ApplicationPropPage.Const_TargetFrameworkMoniker)
+
+            Dim framework = New FrameworkName(CStr(moniker.Value))
+
+            Return provider.GetSupportedTargetFrameworks(framework)
+
+        End Function
+
+        Private Shared Function GetSupportedTargetFrameworksProvider(
+                frameworkMultiTargeting As IVsFrameworkMultiTargeting,
+                project As Project,
+                converter As TypeConverter) As ISupportedTargetFrameworksProvider
+
+            ' Is this a CPS project?
```

Comment:
Would be useful to explain why only CPS provides a non-null converter, and what it's for.